include/domain/acoustic/acoustic_domain.tpp

```diff
@@ -186,7 +187,7 @@ void initialize_sources(
             new (source)
                 source_type<qp_type,
                             specfem::enums::element::property::isotropic>(
-                    ispec, sv_source_array, source_time_function);
+                    ispec, kappa, sv_source_array, source_time_function);
```

Comment:
I think kappa here should be of type_real and not a DeviceView. Since this is an elemental source class it would be better to pass to it the kappa of the element where this source lies. An analogy can be made to `source_array` for example where we pass only a subview of the `source_array` of the element where this source belongs. 

Advantage of doing this would be now if we ever wanted to test elemental source class by itself then we can just pass variables related to that element, resulting in self contained classes. In the current implementation  we'd have to pass kappa for the whole mesh, or at the very list create a scalar 3d view with value of the element - which seems counter intuitative